java/vector/src/main/java/org/apache/arrow/vector/complex/BaseListVector.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.complex;
+
+import org.apache.arrow.vector.FieldVector;
+
+/**
+ * Abstraction for all list type vectors.
+ */
+public interface BaseListVector extends FieldVector {
+
+  /**
+   * Get data vector start index with the given list index.
+   */
+  int getElementStartIndex(int index);
+
+  /**
+   * Get data vector end index with the given list index.
+   */
+  int getElementEndIndex(int index);
+
+  /**
+   * Set data vector for this list vector.
+   */
+  void setDataVector(FieldVector dataVector);
```

Comment:
I think we should use initFromChildren instead of adding this.

Reply:
I think there has a problem without adding this.
Currently in subField, we simply copy ListVector to reuse its validity/offset buffer, and just need to reset it's data vector. If we use initializeChildrenFromFields(List<Field> children), it will throw exception because changing data vector is not allowed here.

Reply:
What we need here is something like https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java#L303
What about remove this method to BaseListVector and make it public?

Reply:
I updated PR as mentioned above, if you have better idea, please let me know, thanks!